flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/WaitingForResourcesTest.java

```diff
@@ -57,10 +57,15 @@ public void testTransitionToExecuting() throws Exception {
         try (MockContext ctx = new MockContext()) {
             ctx.setHasEnoughResources(() -> true);
 
-            WaitingForResources wfr =
-                    new WaitingForResources(ctx, log, RESOURCE_COUNTER, Duration.ZERO);
             ctx.setExpectExecuting(assertNonNull());
-            wfr.onEnter();
+
+            new WaitingForResources(ctx, log, RESOURCE_COUNTER, Duration.ZERO);
+            // run delayed actions
+            for (ScheduledRunnable scheduledRunnable : ctx.getScheduledRunnables()) {
+                if (!ctx.hasStateTransition()) {
```

Comment:
so basically you want to run actions until you hit a state transition?
I think you could make this more obvious by checking for hasStateTransition after running the action and existing the loop